updating links from fwlink to real link if exists (#17091)
dotnet/docs: docs/framework/data/adonet/ef/language-reference/compiled-queries-linq-to-entities.md

  
 Starting with the .NET Framework 4.5, LINQ queries are cached automatically. However, you can still use compiled LINQ queries to reduce this cost in later executions and compiled queries can be more efficient than LINQ queries that are automatically cached. Note that LINQ to Entities queries that apply the `Enumerable.Contains` operator to in-memory collections are not automatically cached. Also parameterizing in-memory collections in compiled LINQ queries is not allowed.  
  

>  The <xref:System.Data.Objects.CompiledQuery> class provides compilation and caching of queries for reuse. Conceptually, this class contains a <xref:System.Data.Objects.CompiledQuery>'s `Compile` method with several overloads. Call the `Compile` method to create a new delegate to represent the compiled query. The `Compile` methods, provided with a <xref:System.Data.Objects.ObjectContext> and parameter values, return a delegate that produces some result (such as an <xref:System.Linq.IQueryable%601> instance). The query compiles once during only the first execution. The merge options set for the query at the time of the compilation cannot be changed later. Once the query is compiled you can only supply parameters of primitive type but you cannot replace parts of the query that would change the generated SQL. For more information, see [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/)  

Reviewer: link here for `Entity Framework Merge Options and Compiled Queries` should point to a different place
-- WOW is the blogs archive site terrible :( no searching allowed. That took me a bit to find...

```suggestion
 The <xref:System.Data.Objects.CompiledQuery> class provides compilation and caching of queries for reuse. Conceptually, this class contains a <xref:System.Data.Objects.CompiledQuery>'s `Compile` method with several overloads. Call the `Compile` method to create a new delegate to represent the compiled query. The `Compile` methods, provided with a <xref:System.Data.Objects.ObjectContext> and parameter values, return a delegate that produces some result (such as an <xref:System.Linq.IQueryable%601> instance). The query compiles once during only the first execution. The merge options set for the query at the time of the compilation cannot be changed later. Once the query is compiled you can only supply parameters of primitive type but you cannot replace parts of the query that would change the generated SQL. For more information, see [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/en-us/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries)  
```

Reviewer 2: It's easy to find them if you use https://wheregoes.com/ with the fwlink. Then I also usually update the fwlink internally when redirect is not properly working. also docs links shouldn't have locales.

Reviewer 2: Please use this suggestion instead @eddynaka 

```suggestion
 The <xref:System.Data.Objects.CompiledQuery> class provides compilation and caching of queries for reuse. Conceptually, this class contains a <xref:System.Data.Objects.CompiledQuery>'s `Compile` method with several overloads. Call the `Compile` method to create a new delegate to represent the compiled query. The `Compile` methods, provided with a <xref:System.Data.Objects.ObjectContext> and parameter values, return a delegate that produces some result (such as an <xref:System.Linq.IQueryable%601> instance). The query compiles once during only the first execution. The merge options set for the query at the time of the compilation cannot be changed later. Once the query is compiled you can only supply parameters of primitive type but you cannot replace parts of the query that would change the generated SQL. For more information, see [EF Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries).
```

>  The <xref:System.Data.Objects.CompiledQuery> class provides compilation and caching of queries for reuse. Conceptually, this class contains a <xref:System.Data.Objects.CompiledQuery>'s `Compile` method with several overloads. Call the `Compile` method to create a new delegate to represent the compiled query. The `Compile` methods, provided with a <xref:System.Data.Objects.ObjectContext> and parameter values, return a delegate that produces some result (such as an <xref:System.Linq.IQueryable%601> instance). The query compiles once during only the first execution. The merge options set for the query at the time of the compilation cannot be changed later. Once the query is compiled you can only supply parameters of primitive type but you cannot replace parts of the query that would change the generated SQL. For more information, see [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries)  

Reviewer 2: ```suggestion
 The <xref:System.Data.Objects.CompiledQuery> class provides compilation and caching of queries for reuse. Conceptually, this class contains a <xref:System.Data.Objects.CompiledQuery>'s `Compile` method with several overloads. Call the `Compile` method to create a new delegate to represent the compiled query. The `Compile` methods, provided with a <xref:System.Data.Objects.ObjectContext> and parameter values, return a delegate that produces some result (such as an <xref:System.Linq.IQueryable%601> instance). The query compiles once during only the first execution. The merge options set for the query at the time of the compilation cannot be changed later. Once the query is compiled you can only supply parameters of primitive type but you cannot replace parts of the query that would change the generated SQL. For more information, see [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries).
```

- [ADO.NET Entity Framework](../index.md)
- [LINQ to Entities](linq-to-entities.md)

> - [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/)

Reviewer: ```suggestion
- [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/en-us/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries)
```

Reviewer 2: Updated your suggestion to have correct link text and no locale

Reviewer: Thanks.. too much copy paste going on in my head hehe

> - [Entity Framework Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries)

Reviewer 2: ```suggestion
- [EF Merge Options and Compiled Queries](https://docs.microsoft.com/archive/blogs/dsimmons/ef-merge-options-and-compiled-queries)
```
